Simple Inventory Management System/Inventory.cs

```diff
@@ -28,5 +28,81 @@ public void DisplayProducts()
             }
         }
 
+        public void EditProduct(string productName)
+        {
+            var product = FindProduct(productName);
+            if (product != null)
+            {
+                DisplayProductDetails(product);
+                UpdateProductDetails(product);
+            }
+            else
+            {
+                Console.WriteLine($"Product with name '{productName}' was not found.");
+            }
+        }
+
+        private Product FindProduct(string productName)
+        {
+            return products.Find(p => p.Name == productName);
+        }
+
+        private void DisplayProductDetails(Product product)
+        {
+            Console.WriteLine($"Product found: {product.Name} - Price: ${product.Price} - Quantity: {product.Quantity}");
+            Console.WriteLine("Enter new details:");
+        }
+
+        private void UpdateProductDetails(Product product)
+        {
+            string newName = GetNewNameFromUser();
+            if (!string.IsNullOrEmpty(newName))
+            {
+                product.Name = newName;
+            }
+
+            decimal newPrice = GetNewPriceFromUser();
+            product.Price = newPrice;
+
+            int newQuantity = GetNewQuantityFromUser();
+            product.Quantity = newQuantity;
+        }
+
+        private string GetNewNameFromUser()
+        {
+            Console.Write("New name : ");
+            return Console.ReadLine();
+        }
+
+        private decimal GetNewPriceFromUser()
+        {
+            decimal newPrice;
+            Console.Write("New price: ");
+            string newPriceStr = Console.ReadLine();
+            while (!decimal.TryParse(newPriceStr, out newPrice))
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("Invalid price format. Please enter a valid price:");
+                Console.ForegroundColor = ConsoleColor.White;
+                newPriceStr = Console.ReadLine();
```

Comment:
Same here, don't force the user to write a new price, what if they want to change the name only? or vice versa